### PR TITLE
fix: remove tooltip rendered element from DOM when is not showing

### LINKED ISSUE #918: [BUG] Tooltips aren't completely hidden on page when not showing
**Describe the bug**

Tooltips are not being fully hidden in the DOM

**Version of Package**
v5.4.0

**To Reproduce**

If you have a list of items (particularly one longer than the view-height of the page), tooltips will take up space in the background even if not active, causing the actual height of the page to be greater than necessary (on our app this caused a huge block of scrollable white space underneath our app).

On inspection, I noticed that tooltips have a CSS property of `visibility: hidden` rather than `display: none`, causing it to take up space (even if invisible).

**Expected behavior**

It should be using `display: none` when the tooltip is not showing to properly remove it from page calculations.

### PR DESCRIPTION
**Pull request by danielbarion**
- [x] fix the tooltip `first show` when the element is not rendered (the position was miscalculated)
- [x] stay with the same behaviors in this PR as before

closes #918

### CODE REVIEW

File: src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -386,13 +401,19 @@ const Tooltip = ({
   }, [])
 
   const hasContentOrChildren = Boolean(html || content || children)
+  const canShow = Boolean(
+    hasContentOrChildren &&
+      !calculatingPosition &&
+      (isOpen || show) &&
+      Object.keys(inlineStyles).length > 0,
```
Comment (danielbarion) [inline src/components/Tooltip/Tooltip.tsx:408]: `Object.keys(inlineStyles).length > 0` this code with the small delay to show the tooltip (time to get the tooltip element rendered before position calculation) fixed the problem of tooltip start showing in a wrong place for the first time

File: src/components/Tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -44,19 +44,34 @@ const Tooltip = ({
...
+     * this `timeout` is necessary because the component
+     * needs to be rendered to calculate the position correctly
+     */
+    setTimeout(() => {
+      if (setIsOpen) {
+        setIsOpen(value)
+      } else if (isOpen === undefined) {
+        setShow(value)
+      }
+    }, 10)
```
Comment (danielbarion) [inline src/components/Tooltip/Tooltip.tsx:74]: needs to double-check the delay tooltip

File: src/components/Tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -44,19 +44,34 @@ const Tooltip = ({
   const [inlineStyles, setInlineStyles] = useState({})
   const [inlineArrowStyles, setInlineArrowStyles] = useState({})
   const [show, setShow] = useState(false)
+  const [rendered, setRendered] = useState(false)
```
Comment (danielbarion) [inline src/components/Tooltip/Tooltip.tsx:47]: added a new prop to handle if the tooltip can be rendered or not, this way we can handle the animations and the element render

File: src/components/Tooltip/Tooltip.tsx [hunk 4]
```diff
@@ -44,19 +44,34 @@ const Tooltip = ({
...
-  const handleShow = (value: boolean) => {
-    if (setIsOpen) {
-      setIsOpen(value)
-    } else if (isOpen === undefined) {
-      setShow(value)
+  useEffect(() => {
+    if (!show) {
+      setRendered(false)
     }
+  }, [show])
```
Comment (danielbarion) [inline src/components/Tooltip/Tooltip.tsx:59]: need double-check the delay tooltip

**Comment by danielbarion** [comment]
Test snapshots not updated yet

File: src/components/Tooltip/Tooltip.tsx (hunk 2 above)
Comment (gabrieljablonski) [reply src/components/Tooltip/Tooltip.tsx:73]: Ideally we shouldn't use a timeout, and actually wait for the styles to be updated. 

I couldn't see weird behaviors, so it should be fine for now, but it's something to think about.

**Comment by gabrieljablonski** [comment]
Tests will need to be fixed, since they assumed the tooltip was always in the DOM.

After that, this should be good to merge.

**Review by gabrieljablonski — APPROVED** [review]
The tests are simple enough that we probably should be fine for now without them.

**Pull request merged**